core/src/com/mygdx/game/client/screen/ChooseUnitFieldDialog.java

```diff
@@ -0,0 +1,58 @@
+package com.mygdx.game.client.screen;
+
+import com.badlogic.ashley.core.ComponentMapper;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Dialog;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.mygdx.game.client.component.FieldComponent;
+import com.mygdx.game.client.component.UnitComponent;
+import lombok.extern.java.Log;
+
+import java.util.logging.Level;
+
+@Log
+public class ChooseUnitFieldDialog {
+
+  private static final ComponentMapper<FieldComponent> fieldComponentMapper = ComponentMapper.getFor(FieldComponent.class);
+  private static final ComponentMapper<UnitComponent> unitComponentMapper = ComponentMapper.getFor(UnitComponent.class);
+  private static final Skin uiSkin = new Skin(Gdx.files.internal("dialog/custom/uiskin.json"));
+
+  public static Dialog customDialog = null;
+
+  public static void createCustomDialog(Stage parentStage, Entity entity) {
+    if (customDialog != null) {
+      customDialog.hide();
+      customDialog = null;
+    }
+
+
+    customDialog = new Dialog("Choose", uiSkin) {
+      @Override
+      protected void result(Object object) {
```

Comment:
ale ten libgdx jest kiepski xD

callbacków, ani nic nie ma i trzeba robić takie dziedziczenie + w ogóle brak jakiegoś generyka

ale to jest wina właśnie libgdxa